src/providers/ad/ad_gpo.c

```diff
@@ -1866,6 +1867,8 @@ ad_gpo_access_send(TALLOC_CTX *mem_ctx,
      */
     state->user_domain = domain;
     state->host_domain = get_domains_head(domain);
+    state->ad_domain = dp_opt_get_string(ctx->ad_id_ctx->ad_options->basic,
```

Comment:
Should we handle situation when NULL is returned as ad_domain here?

Reply:
This is not needed since this option is always set (defaults to sssd domain name). See: [src/providers/ad/ad_common.c:511](src/providers/ad/ad_common.c:511)